### PR TITLE
Prepare setup for runnign e2e tests

### PR DESCRIPTION
**Pull request by krhubert**
This PR adds possibilities to run e2e with `make` and run them in circleci
close https://github.com/mesg-foundation/engine/issues/1418
close https://github.com/mesg-foundation/engine/issues/1417
close https://github.com/mesg-foundation/engine/issues/1419

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -36,16 +36,15 @@ save_go_cache: &save_go_cache
...
     environment:
       - GOCACHE: *go_cache_dir
     steps:
       - checkout
-      - setup_remote_docker
       - run: docker swarm init
       - <<: *restore_go_cache
       - <<: *restore_go_mod_cache
-      - run: make test
+      - run: make test e2e
```
[the remark this answers is not in the document]
Comment (NicolasMahe) [reply .circleci/config.yml]: Done in https://github.com/mesg-foundation/engine/pull/1436

File: scripts/dev.sh [hunk 2]
```diff
@@ -37,31 +37,66 @@ function docker_network_remove {
...
+    mesg/engine:local
+}
+
+function stop_engine {
+  onexit
+}
+
+quiet=false
+
+while getopts "q" o; do
```
Comment (NicolasMahe) [inline scripts/dev.sh]: could we use a nicer and simpler system to parse flags? maybe there is a bash lib that can do it?

File: scripts/dev.sh [hunk 3]
```diff
@@ -37,31 +37,66 @@ function docker_network_remove {
...
+    q)
+      quiet=true
+      ;;
+    *)
+      echo "unknown flag $0"
+      exit 1
+      ;;
+  esac
+done
+shift $((OPTIND-1))
```
Comment (NicolasMahe) [inline scripts/dev.sh]: what?

**Comment by NicolasMahe** [comment]
I fix the CI on PR https://github.com/mesg-foundation/engine/pull/1436

File: Makefile [hunk 4]
```diff
@@ -39,14 +39,26 @@ docker-tools:
...
+	./scripts/dev.sh stop
+
 dep:
 	go mod download
 
 build: check-version dep
 	go build -mod=readonly -o ./bin/engine -ldflags="-X 'github.com/mesg-foundation/engine/version.Version=$(version)'" core/main.go
 
+e2e: export MESG_PATH = $(PWD)/e2e.test/mesg
+e2e: clean-e2e
```
[the remark this answers is not in the document]
Comment (krhubert) [reply Makefile]: Yes. That syntax is used to export env vars. And the MESG_PATH is changed to have a dedicated path for e2e tests

File: scripts/dev.sh (hunk 2 above)
Comment (krhubert) [reply scripts/dev.sh]: Yes, there is a bash tool and it is called .... getopts. This is only avaiable on linux and mac (installed by default) and it is pretty common and stanard.

File: scripts/dev.sh (hunk 3 above)
Comment (krhubert) [reply scripts/dev.sh]: It's bash default syntax to remove element from array. In that case it removes element from arguments array.

**Comment by NicolasMahe** [comment]
@krhubert can you clean the git history (there is a WIP commit) before merge

**Pull request merged**